kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -56,240 +58,245 @@
  * {@link RowToInsert RowToInserts} and subsequently write them to BigQuery.
  */
 public class BigQuerySinkTask extends SinkTask {
-  private static final Logger logger = LoggerFactory.getLogger(BigQuerySinkTask.class);
-
-  private final BigQuery testBigQuery;
-  private SchemaRetriever schemaRetriever;
-  private BigQueryWriter bigQueryWriter;
-  private BigQuerySinkTaskConfig config;
-  private RecordConverter<Map<String, Object>> recordConverter;
-  private Map<String, TableId> topicsToBaseTableIds;
-  private boolean useMessageTimeDatePartitioning;
-
-  private TopicPartitionManager topicPartitionManager;
-
-  private KCBQThreadPoolExecutor executor;
-  private static final int EXECUTOR_SHUTDOWN_TIMEOUT_SEC = 30;
-
-  public BigQuerySinkTask() {
-    testBigQuery = null;
-    schemaRetriever = null;
-  }
-
-  // For testing purposes only; will never be called by the Kafka Connect framework
-  public BigQuerySinkTask(BigQuery testBigQuery, SchemaRetriever schemaRetriever) {
-    this.testBigQuery = testBigQuery;
-    this.schemaRetriever = schemaRetriever;
-  }
-
-  @Override
-  public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
-    try {
-      executor.awaitCurrentTasks();
-    } catch (InterruptedException err) {
-      throw new ConnectException("Interrupted while waiting for write tasks to complete.", err);
-    }
-    updateOffsets(offsets);
-
-    topicPartitionManager.resumeAll();
-  }
-
-  /**
-   * This really doesn't do much and I'm not totally clear on whether or not I need it.
-   * But, in the interest of maintaining old functionality, here we are.
-   */
-  private void updateOffsets(Map<TopicPartition, OffsetAndMetadata> offsets) {
-    for (Map.Entry<TopicPartition, OffsetAndMetadata> offsetEntry : offsets.entrySet()) {
-      context.offset(offsetEntry.getKey(), offsetEntry.getValue().offset());
+    private static final Logger logger = LoggerFactory.getLogger(BigQuerySinkTask.class);
+
+    private final BigQuery testBigQuery;
+    private SchemaRetriever schemaRetriever;
+    private BigQueryWriter bigQueryWriter;
+    private BigQuerySinkTaskConfig config;
+    private RecordConverter<Map<String, Object>> recordConverter;
+    private Map<String, TableId> topicsToBaseTableIds;
+    private boolean useMessageTimeDatePartitioning;
+    private boolean usePartitionByCustomKey;
```

Comment:
rename as useCustomKeyPartitioning to unify the naming format?

Reply:
Will do!!!